drivers/hmis/app/graphql/types/hmis_schema/assessment.rb

```diff
@@ -39,26 +39,65 @@ class HmisSchema::Assessment < Types::BaseObject
     field :health_and_dv, Types::HmisSchema::HealthAndDv, null: true
     field :exit, Types::HmisSchema::Exit, null: true
     field :disability_group, Types::HmisSchema::DisabilityGroup, null: true
+    field :youth_education_status, Types::HmisSchema::YouthEducationStatus, null: true
+    field :employment_education, Types::HmisSchema::EmploymentEducation, null: true
     custom_data_elements_field
 
+    field :role, Types::Forms::Enums::AssessmentRole, null: false
+    field :definition, Types::Forms::FormDefinition, null: false
+    field :wip_values, JsonObject, null: true
+
+    def wip_values
+      return unless object.in_progress?
+
+      load_ar_association(object, :form_processor)&.values
+    end
+
+    def role
+      Hmis::Form::Definition::FORM_DATA_COLLECTION_STAGES.invert[object.data_collection_stage]&.to_s
+    end
+
+    # EXPENSIVE! Do not use in batch
+    def definition
+      # If definition is stored on form processor, return that
+      # TODO: check if form is retired? For non-WIP assessments, we should
+      # really be choosing the "latest" form, which may not be the one on the FormProcessor.
+      form_processor = load_ar_association(object, :form_processor)
+      definition = load_ar_association(form_processor, :definition)
```

Comment:
Ok, this looks like how we're determining the version.  Basically, use the one we have, if we have one, otherwise the current version.  I think I like that.

Reply:
Is there a situation where the form definition ID might have changed (as in someone associated a completely different form definition with this assessment?)  We might want to check for that in addition to stale-ness if that's possible.)

Reply:
yes, exactly, thats what my comment says :) it is not handled yet.
I hesitated to ALWAYS look up the latest form for submitted assessments, because of the additional performance hit